src/transformers/models/mplug_owl/processing_mplug_owl.py

```diff
@@ -0,0 +1,150 @@
+    image_processor_class = "CLIPImageProcessor"
+    tokenizer_class = "LlamaTokenizer"
+
+    def __init__(self, image_processor=None, tokenizer=None, **kwargs):
+        super().__init__(image_processor, tokenizer)
+        self.tokens_to_generate = 0
+        self.add_BOS = True
+
+    def __call__(self, text=None, images=None, return_tensors=None, **kwargs):
+        if text is None and images is None:
+            raise ValueError("You have to specify either text or images. Both cannot be none.")
+
+        if text is not None:
+            encoding = self.tokenize_prompts(
+                prompts=text,
+                tokens_to_generate=self.tokens_to_generate,
+                add_BOS=self.add_BOS,
+                tokenizer=self.tokenizer,
+                ignore_dist=True,
+                **kwargs,
+            )
+
+        if images is not None:
+            image_features = self.image_processor(images, return_tensors=return_tensors, **kwargs)
+
+        if text is not None and images is not None:
+            encoding["pixel_values"] = image_features.pixel_values
+            return BatchEncoding(data=encoding)
+        elif text is not None:
+            return BatchEncoding(data=encoding)
+        else:
+            return BatchEncoding(data=dict(**image_features), tensor_type=return_tensors)
+
+    def batch_decode(self, skip_special_tokens=True, *args, **kwargs):
+        """
+        This method forwards all its arguments to CLIPTokenizerFast's [`~PreTrainedTokenizer.batch_decode`]. Please
+        refer to the docstring of this method for more information.
+        """
+        return self.tokenizer.batch_decode(*args, skip_special_tokens=skip_special_tokens, **kwargs)
+
+    def decode(self, skip_special_tokens=True, *args, **kwargs):
+        """
+        This method forwards all its arguments to CLIPTokenizerFast's [`~PreTrainedTokenizer.decode`]. Please refer to
+        the docstring of this method for more information.
+        """
+        return self.tokenizer.decode(*args, skip_special_tokens=skip_special_tokens, **kwargs)
+
+    def tokenize_prompts(self, prompts=None, tokens_to_generate=None, add_BOS=None, rank=0, **kwargs):
+        """Tokenize prompts and make them avaiable on all ranks."""
+
+        # On all ranks set to None so we can pass them to functions
+        prompts_tokens_cuda_long_tensor = None
+        prompts_length_cuda_long_tensor = None
+
+        # On the specified rank, build the above.
+        attention_mask = None
+
+        (
+            prompts_tokens_cuda_long_tensor,
+            prompts_length_cuda_long_tensor,
+            attention_mask,
+        ) = self._tokenize_prompts_and_batch(prompts, tokens_to_generate, add_BOS, **kwargs)
+
+        return {
+            "input_ids": prompts_tokens_cuda_long_tensor,
+            "attention_mask": attention_mask,
+        }
+
+    def _tokenize_prompts_and_batch(self, prompts, tokens_to_generate, add_BOS, **kwargs):
+        import torch
```

Comment:
you can add a `if_torch_avaialble` and import it on top of the file, check: https://github.com/huggingface/transformers/blob/3658488ff77ff8d45101293e749263acf437f4d5/src/transformers/models/sam/processing_sam.py#LL28C1-L29C17